### PR TITLE
feat: added `Create` method 

### PR DESCRIPTION
**Pull request by mattia-fumo**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -157,3 +157,22 @@ func (c Client[Resource]) List(ctx context.Context, options Options) ([]Resource
...
+	req, err := c.client.NewRequestWithContext(ctx, http.MethodPost, "", resource)
+	if err != nil {
+		return "", err
+	}
+
+	if err := options.setOptionsInRequest(req); err != nil {
+		return "", err
+	}
+
+	createdResource := make(map[string]string)
```
Comment (davidebianchi) [inline client.go:173]: Can you define a struct of the expected response?

File: client_test.go [hunk 2]
```diff
@@ -504,6 +505,77 @@ func TestList(t *testing.T) {
...
+		require.Equal(t, expectedID, actualID)
+	})
+
+	t.Run("throws - bad request", func(t *testing.T) {
+		testhelper.NewGockScope(t, baseURL, http.MethodPost, "").
+			BodyString(string(expectedBody)).
+			Reply(400).
+			JSON(CrudErrorResponse{
+				Message:    "missing required field",
+				StatusCode: 404,
```
Comment (davidebianchi) [inline client_test.go:545]:
```suggestion
				StatusCode: 400,
```

File: client_test.go [hunk 3]
```diff
@@ -504,6 +505,77 @@ func TestList(t *testing.T) {
...
+				Error:      "Bad Request",
+			})
+
+		actualID, err := client.Create(ctx, resourceToCreate, Options{})
+		require.EqualError(t, err, "missing required field")
+		require.Empty(t, actualID)
+	})
+
+	t.Run("proxy headers in request", func(t *testing.T) {
+
```
Comment (davidebianchi) [inline client_test.go:555]:
```suggestion
```

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (mattia-fumo) [reply client.go:173]: Of course!

File: client_test.go [hunk 4]
```diff
@@ -504,6 +505,76 @@ func TestList(t *testing.T) {
...
+		Nested: NestedResource{
+			Field: "something",
+		},
+	}
+
+	expectedBody, err := json.Marshal(resourceToCreate)
+	require.NoError(t, err)
+
+	t.Run("creates resource", func(t *testing.T) {
+
```
Comment (davidebianchi) [inline client_test.go:526]:
```suggestion
```

**Review by davidebianchi — CHANGES_REQUESTED** [review]

**Review by davidebianchi — APPROVED** [review]

**Pull request merged**